runtime/proto/corfu_store_metadata.proto

```diff
@@ -30,8 +30,12 @@ message TableMetadata {
     TableName table_name = 1;
     // True if the CorfuStore is disk based and False if it can be contained in memory.
     bool disk_based = 2;
+
+    // Full path to disk based location if this table is disk based
+    string disk_based_path = 3;
```

Comment:
The metadata is for a distributed object, but the path of is local information. Does it make sense to store it in this registry ?

Reply:
Technically true, but in practice all distributed clients pass the same path. 
In any way we require a way to communicate the location of the disk path from the client's table to the checkpointer so the latter can use the exact same path as a prefix, otherwise we risk in IO overload if the disk path points to a separate disk.

Reply:
I see.

But what happens if the client and the checkpointer run on the same machine. Wouldn't they both point to the same path? 

Since during open the DB in the destination path is destroyed https://github.com/CorfuDB/CorfuDB/blob/master/runtime/src/main/java/org/corfudb/runtime/collections/PersistedStreamingMap.java#L54 , this will crash one of the clients.

Reply:
This path location added in this PR is info-only. It is up to the clients to ensure that they do not pass conflicting paths and open the same table from multiple clients. For example the checkpointer client will suffix the table with a subfolder like _compactor_tablename/ to ensure that it does not conflict. We can at best add a few comments explaining this.